core/services/vrf/listener_v2.go

```diff
@@ -321,141 +384,251 @@ func (lsn *listenerV2) processRequestsPerSub(
-			processed[vrfRequest.RequestId.String()] = struct{}{}
-			continue
+			processed[req.req.RequestId.String()] = struct{}{}
+		} else {
+			unconsumed = append(unconsumed, req)
 		}
+	}
 
-		// Check if we can ignore the request due to it's age.
-		if time.Now().UTC().Sub(req.utcTimestamp) >= lsn.job.VRFSpec.RequestTimeout {
-			rlog.Infow("Request too old, dropping it")
-			lsn.markLogAsConsumed(req.lb)
-			processed[vrfRequest.RequestId.String()] = struct{}{}
-			incDroppedReqs(lsn.job.Name.ValueOrZero(), lsn.job.ExternalJobID, v2, reasonAge)
-			continue
+	// Process requests in chunks
+	for chunkStart := 0; chunkStart < len(unconsumed); chunkStart += int(lsn.job.VRFSpec.ChunkSize) {
+		chunkEnd := chunkStart + int(lsn.job.VRFSpec.ChunkSize)
+		if chunkEnd > len(unconsumed) {
+			chunkEnd = len(unconsumed)
 		}
-
-		// Check if the vrf req has already been fulfilled.
-		// Pass max(request.BlockNumber, currentBlockNumber) to the call to ensure we get
-		// the latest state or an error.
-		// This is to avoid a race condition where a request commitment has been written on-chain
-		// but has not been synced fully on the entire network. GetCommitment will return an empty
-		// byte array in that case, and we would think that the request is fulfilled when it actually
-		// isn't.
-		reqBlockNumber := new(big.Int).SetUint64(vrfRequest.Raw.BlockNumber)
-		currBlock := new(big.Int).SetUint64(lsn.getLatestHead())
-		m := bigmath.Max(reqBlockNumber, currBlock)
-		callback, err := lsn.coordinator.GetCommitment(&bind.CallOpts{
-			BlockNumber: m,
-		}, vrfRequest.RequestId)
-		if err != nil {
-			rlog.Warnw("Unable to check if already fulfilled, processing anyways",
-				"err", err,
-				"reqBlockNumber", reqBlockNumber,
-				"currBlock", currBlock,
-			)
-		} else if utils.IsEmpty(callback[:]) {
-			// If seedAndBlockNumber is zero then the response has been fulfilled
-			// and we should skip it
-			rlog.Infow("Request already fulfilled", "callback", callback)
-			lsn.markLogAsConsumed(req.lb)
-			processed[vrfRequest.RequestId.String()] = struct{}{}
-			continue
+		chunk := unconsumed[chunkStart:chunkEnd]
+
+		var unfulfilled []pendingRequest
+		alreadyFulfilled := lsn.checkReqsFulfilled(l, chunk)
+		for i, a := range alreadyFulfilled {
+			if a {
+				lsn.markLogAsConsumed(chunk[i].lb)
+				processed[chunk[i].req.RequestId.String()] = struct{}{}
+			} else {
+				unfulfilled = append(unfulfilled, chunk[i])
+			}
 		}
-		// Run the pipeline to determine the max link that could be billed at maxGasPrice.
-		// The ethcall will error if there is currently insufficient balance onchain.
-		maxLink, run, payload, gaslimit, err := lsn.getMaxLinkForFulfillment(maxGasPriceWei, req, rlog)
+
+		fromAddress, err := lsn.gethks.GetRoundRobinAddress(lsn.chainID, lsn.fromAddresses()...)
 		if err != nil {
-			rlog.Warnw("Unable to get max link for fulfillment, skipping request", "err", err)
+			l.Errorw("Couldn't get next from address", "err", err)
 			continue
```

Comment:
Kinda not related to this PR, but ... the two possible errors that GetRoundRobinAddress returns are:

* keystore locked
* no sending keys available

In both cases there's not much we can do. I'm wondering if we should be louder here and just panic, because if we ever get here, it's a misconfiguration, and better to fail loud than silent in this situation IMO. But this could be a separate PR.

Reply:
Yeah..I think how to deal with fatal-ish errors is kinda something unsolved in the core node. I think panicing in  this case could make the issue worse because then the node crashes and you can't actually fix the config (by adding keys, unlocking keystore etc) since the node has to be up for the API to work